components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -2207,3 +2208,34 @@ func validateFiltersTimeRange(endTime string, startTime string) error {
 	}
 	return nil
 }
+
+func getStartDateFromEndDate(endTime string, startTime string) ([]string, error) {
+	if len(endTime) == 0 {
+		return nil, nil
+	}
+
+	parsedEndTime, err := time.Parse(time.RFC3339, endTime)
+	if err != nil {
+		return []string{}, err
+	}
+
+	if checkTodayIsEndTime(parsedEndTime) {
+		if startTime == "" {
+			return []string{}, nil
+		}
+		return []string{startTime}, nil
```

Comment:
If the input is 
start date:= 10th July 
end date := 29th July (current date)

this will make start date as current date.
That is not correct. Please confirm

Reply:
No, if the end date is current day, It will check and then return the start time as either blank or the start time that was there in the request.

Reply:
ah. Missed it.